### PR TITLE
[Duplicate] Allow custom properties when duplicating an object

### PR DESCRIPTION
**Pull request by BogdanAlexandru**
This almost solves #604. Custom properties are settable, but it lacks proper test coverage, some cleanup is needed, comments need adding/changing and any other changes that come up during this review. 
It can be seen in action in [**this gif**](https://gfycat.com/DifferentRipeLadybird).

Since I had most of the changes available locally, I went ahead and got this out of the way so as to not hold the issue any longer.

See the original issue for the reasoning behind my decisions. There are some things I want to ask about especially:

1) I added a separate CopyActionWizard to keep custom properties input simple. This wizard is almost 100% identical to the CreateWizard. The two have different purposes (in that they're used for different purposes, but both create a model), but finding a way to merge them together would get rid of that extra Spec file. Any thoughts on what the best approach here would be?
A `ModelBuildingWizard` maybe, with the sole purpose of taking user input and building a model with it? That could cover both cases successfully I think.

2) Duplicate always points to the root of the tree initially when picking a location (and I think it always did this). Do we want to point to the current parent instead?

### CODE REVIEW

**Comment by BogdanAlexandru** [comment]
Bumping for points 1) and 2), once an approach is chosen for them I can go ahead and finish this up.

File: platform/entanglement/src/actions/CopyAction.js [hunk 1]
```diff
@@ -103,10 +102,57 @@ define(
...
+                newContext.selectedObject = object;
+                newContext.domainObject = newParentObj;
+                return copyService.validate(object, newParentObj) &&
+                    policyService.allow("action", self, newContext);
+            };
+
+            return new CopyActionWizard(this.object, undefined, validateLocation);
+        };
+
+        CopyAction.prototype.performBase = function () {
```
Comment (akhenry) [inline platform/entanglement/src/actions/CopyAction.js]: The splitting up of `perform` and `performBase` feels a little arbitrary (I understand that it reflects that fact that this class previously extended another). I'd prefer it these were combined into one function, and then `notification`, `success`, and `error` could be broken out into member functions of the `CopyAction` in order to tidy the function up a bit. You can use the @private annotation in jsdocs to indicate that they are private functions.

File: platform/entanglement/src/actions/CopyActionWizard.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+define(
+    function () {
+        function CopyActionWizard(domainObject, parent, locationValidator) {
```
Comment (akhenry) [inline platform/entanglement/src/actions/CopyActionWizard.js]: This wizard is almost 100% identical to the CreateWizard. The two have different purposes (in that they're used for different purposes, but both create a model), but finding a way to merge them together would get rid of that extra Spec file. 

Yeah, I agree. Let's see if we can merge them. In fact, the [`PropertiesDialog`](https://github.com/nasa/openmct/blob/master/platform/commonUI/edit/src/actions/PropertiesDialog.js) is also basically the same except for the location selector. Perhaps we could use the `PropertiesDialog` as a base class and extend it with a `NewObjectDialog` that includes the location control. Then we can use the `NewObjectDialog` from the CreateAction and the `CopyAction`. Thoughts?

CreateWizard and CopyActionWizard also have a `createModel` function on them. It basically does the same thing as `updateModel` on `PropertiesDialog` so I'd be inclined to get rid of `createModel` altogether and just externalize the instantiation of the new model object and pass it into `updateModel`.

**Comment by BogdanAlexandru** [comment]
Yeah, it would be best if someone else continued this, there's no way I can work reliably on it right now.

**Comment by akhenry** [comment]
@BogdanAlexandru No worries, thanks for letting us know.

**Comment by larkin** [comment]
Going to close this as we don't have resources to push this forward and we're also working on a large set of refactors that may substantially change how it is resolved.  That said, big thanks to @BogdanAlexandru, we appreciate your efforts!

**Pull request closed without merging**